src/Sylius/Bundle/CoreBundle/PaymentRequest/Checker/PaymentRequestIntegrityChecker.php

```diff
@@ -15,9 +15,10 @@
 
 use Sylius\Bundle\CoreBundle\PaymentRequest\Command\PaymentRequestHashAwareInterface;
 use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Exception\NullGatewayConfigException;
+use Sylius\Component\Payment\Exception\PaymentRequestNotFoundException;
 use Sylius\Component\Payment\Model\PaymentRequestInterface;
 use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
-use Webmozart\Assert\Assert;
 
 final class PaymentRequestIntegrityChecker implements PaymentRequestIntegrityCheckerInterface
```

Comment:
This checker check two things:
- That `$paymentRequest` exists
- That `paymentMethod` has any `GatewayConfig`

Hence IMO this checker needs to be deeply refactored or at least renamed to something like `PaymentMethodGatewayConfigExistsChecker`.

Reply:
https://github.com/Sylius/Sylius/pull/15979#discussion_r1521426780